drivers/flash/spi_nor.c

```diff
@@ -1254,12 +1279,37 @@ static int spi_nor_configure(const struct device *dev)
  */
 static int spi_nor_init(const struct device *dev)
 {
+#if (ANY_INST_HAS_WP_GPIOS || ANY_INST_HAS_HOLD_GPIOS)
+	const struct spi_nor_config *cfg = dev->config;
+#endif
+
 	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
 		struct spi_nor_data *const driver_data = dev->data;
 
 		k_sem_init(&driver_data->sem, 1, K_SEM_MAX_LIMIT);
 	}
 
+#if ANY_INST_HAS_WP_GPIOS
+	if (cfg->wp) {
+		if (!device_is_ready(cfg->wp->port)) {
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(cfg->wp, GPIO_OUTPUT_ACTIVE)) {
+			return -ENODEV;
+		}
+	}
+#endif /* ANY_INST_HAS_WP_GPIOS */
+#if ANY_INST_HAS_HOLD_GPIOS
+	if (cfg->hold) {
+		if (!device_is_ready(cfg->hold->port)) {
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(cfg->hold, GPIO_OUTPUT_INACTIVE)) {
+			return -ENODEV;
+		}
+	}
+#endif /* ANY_INST_HAS_HOLD_GPIOS */
```

Comment:
At least some LOG_DBG is needed here. There are now four (probably additional) ways for the code to return `-ENODEV`, which require engaging debugger to figure out.

Reply:
@de-nordic It is an unexpected error state, would LOG_ERR be OK?

Reply:
OK, so there is no way for caller to report this, as `z_sys_init_run_level` seems to log nothing, this means that here is the only place the problem can be described by log; assuming that a device is full tested and some design problems have been weed out, this problems will only happen in field. This means LOG_DBG may require pulling board of the field or uploading DBG software, where uploading debug software may not be possible in case when DFU happens to external flash.
Yes. I think this should be LOG_ERR. Question is whether we can save some flash by reducing the LOG_ERR. Text does not compile that well.